File: react/Services/ViewerService.ts
import axios from "axios";
import CouponItem from "../Models/Coupon/CouponItem";

class ViewerService {
	public async getAllCoupons(){
        return (await axios.get<CouponItem[]>("http://localhost:8080/api/public/coupons")).data;
    }
}
const viewerService = new ViewerService();
export default viewerService;

File: webApp/src/main/java/com/webApp/services/ViewerService.java
package com.webApp.services;

import com.webApp.models.Coupon;
import com.webApp.exceptions.client.ClientDoesNotExistException;
import com.webApp.exceptions.coupon.*;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ViewerService extends ClientService {
    /**
     *
     * @return a list of all coupons.
     */
    public List<Coupon> getAllCoupons(){
        return couponRepository.findAll();
    }
}
